src/main/java/org/skife/jdbi/v2/ClasspathStatementLocator.java

```diff
@@ -118,24 +119,30 @@ public String locate(String name, StatementContext ctx)
                 found.put(cache_key, name == cache_key ? new String(name) : name);
                 return name;
             }
-
-            final StringBuffer buffer = new StringBuffer();
-            reader = new BufferedReader(new InputStreamReader(in_stream, Charset.forName("UTF-8")));
-            String line;
+            StringBuilder sb = new StringBuilder();
             try {
-                while ((line = reader.readLine()) != null) {
-                    if (isComment(line)) {
-                        // comment
-                        continue;
+                SqlScriptLexer lexer = new SqlScriptLexer(new ANTLRInputStream(in_stream));
+                while (true) {
+                    Token t = lexer.nextToken();
+                    int type = t.getType();
+                    if (type == SqlScriptLexer.EOF) {
```

Comment:
This if construct is dodgy, if new token types are introduced they could be silently ignored.  Perhaps either a `switch` statement, or maybe move the big `else if` into a generic `else` block.